streaming/replication.c

```diff
@@ -0,0 +1,1931 @@
+                return rrdpush_receiver_permission_denied(w);
+            }
+            simple_pattern_free(machine_allow_from);
+        }
+    }
+    //TODO: Remove the redundant parameters
+    UNUSED(update_every);
+    UNUSED(abbrev_timezone);
+    UNUSED(tags);
+    UNUSED(timezone);
+    UNUSED(os);                
+    UNUSED(utc_offset);
+    UNUSED(registry_hostname);                 
+
+    // Replication request rate limit control
+    if(unlikely(web_client_replication_rate_t > 0)) {
+        static netdata_mutex_t replication_rate_mutex = NETDATA_MUTEX_INITIALIZER;
+        static volatile time_t last_replication_accepted_t = 0;
+
+        netdata_mutex_lock(&replication_rate_mutex);
+        time_t now = now_realtime_sec();
+
+        if(unlikely(last_replication_accepted_t == 0))
+            last_replication_accepted_t = now;
+
+        if(now - last_replication_accepted_t < web_client_replication_rate_t) {
+            netdata_mutex_unlock(&replication_rate_mutex);
+            rrdhost_system_info_free(system_info);
+            error("%s [receive from [%s]:%s]: too busy to accept new streaming request. Will be allowed in %ld secs.", REPLICATION_MSG, w->client_ip, w->client_port, (long)(web_client_replication_rate_t - (now - last_replication_accepted_t)));
+            return rrdpush_receiver_too_busy_now(w);
+        }
+
+        last_replication_accepted_t = now;
+        netdata_mutex_unlock(&replication_rate_mutex);
+    }
+
+    // TBR - Comment
+    // What it does: if host doesn't exist prepare the receiver state struct
+    // and start the streaming receiver to create it.
+    // What I want:  If the host doesn't exist I should depend on streaming to create it.
+    // At this point, streaming should have already call the receiver thread to create the host.
+    // So if the host exists we continue with the call to the replication rx thread.
+    // If the host doesn't exist and host->receiver is NULL means that there was a problem
+    // with host creation during streaming or the REPLICATE command arrived earlier than the 
+    // respective STREAM command. So do not start the Rx replication thread.
+    // The replication Tx thread in child should try to reconnect.
+
+    rrd_rdlock();
+    RRDHOST *host = rrdhost_find_by_guid(machine_guid, 0);
+    if (unlikely(host && rrdhost_flag_check(host, RRDHOST_FLAG_ARCHIVED))) /* Ignore archived hosts. */
+        host = NULL;
+    if(!host) {
+            rrd_unlock();
+            log_replication_connection(w->client_ip, w->client_port, key, machine_guid, hostname, "ABORT REPLICATION - HOST DOES NOT EXIST");
+            infoerr("%s - [received from [%s]:%s]: Host(%s) with machine GUID %s does not exist - Abort replication.", REPLICATION_MSG, w->client_ip, w->client_port, hostname, machine_guid);
+            return 409;
```

Comment:
Inside [web_client.h](https://github.com/netdata/netdata/blob/8b0fffe33bccf950496f5ffee358320bde71e5b5/web/server/web_client.h#L13-L28) we have defined the codes we are using with specific `defines`, I suggest we also create another for this `409`, because it is also defined for HTTP protocol as you can see [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409).

Reply:
This block is also using a different tabulation, please, bring it for our pattern.